kumascript/src/api/util.js

```diff
@@ -170,17 +170,26 @@ class HTMLTool {
     // If all else, generate a unique one.
     $([...INJECT_SECTION_ID_TAGS].join(",")).each((i, element) => {
       const $element = $(element);
+      const isDt = $element[0].name === "dt";
       // Default is the existing one. Let's see if we need to change it.
       let id = $element.attr("id");
       if ($element.attr("name")) {
         // The "name" attribute overrides any current "id".
         id = slugify($element.attr("name"));
       } else if (id) {
         // If it already has an ID, respect it and leave it be.
-      } else if (H1_TO_H6_TAGS.has($element[0].name)) {
-        // For heading tags, we'll give them an "id" that's a
+      } else if (H1_TO_H6_TAGS.has($element[0].name) || isDt) {
+        // For heading and dt tags, we'll give them an "id" that's a
         // slugified version of their text content.
-        const text = $element.text();
+        let text = $element.text();
+        if (isDt) {
+          // dt elements can, along with the actual term, contain stuff
+          // like <span class="badge inline optional">Optional</span>. If
+          // we don’t trim that, we end up with generated IDs like
+          // id="rtcSessionDescriptionInit_Optional". So, if the dt content
+          // contains a space, we take just whatever precedes the space.
+          text = text.split(" ")[0];
```

Comment:
One thing I've seen in a few places is where there isn't a space before the inline macro, like: https://github.com/mdn/content/blob/0725507c36a4e98f7d01e98f6ef0bb0559f72ff8/files/en-us/web/api/document/index.md?plain=1#L122 :

```
- {{DOMxRef("Document.URL")}}{{ReadOnlyInline}}
  - : Returns the document location as a string.
```
Then the ID ends up like `document.urlread`:

```html
<dt id="document.urlread">
  <a href="/en-US/docs/Web/API/Document/URL">
    <code>Document.URL</code>
  </a>
  <span title="This value may not be changed." class="badge inline readonly">Read only </span>
</dt>
```

It's not ideal, but maybe it's OK, and the markup here is incorrect of course. And alternative ways of getting the text out might have unexpected side effects too.

Reply:
> One thing I've seen in a few places is where there isn't a space before the inline macro, … Then the ID ends up like `document.urlread`:
>
> It's not ideal, but maybe it's OK, and the markup here is incorrect of course.

Thanks much for catching that. I think it’s essential that we may it do the right thing even in case of suboptimal markup. I guess arguably that markup isn’t outright incorrect — since it doesn’t cause any rendering problems (right?).

Anyway, in hindsight I can see that taking the text content before the first space was a brittle hack. So I spent some time today reading up on the cheerio API, and found what I think is the right way to ensure we always get just the text content of the first element child of the `dt`. So I’ve pushed a change which does it that way instead.